### PR TITLE
[ASCII-1081] Remove Stop method from demultiplexer interface

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?
This PR removes the `Stop` method from the demultiplexer interface in favour of using fx `OnStop` hook.

This PR must be merged after https://github.com/DataDog/datadog-agent/pull/22044 to make sure `OnStop` on the demultiplexer instance is called even when there is an error.

### Describe how to test/QA your changes
Run the command (see below), wait 3-4 seconds (should be less than 10s) and stop the command with `ctrl+c` .  Check `Stopping aggregator` is in the log of the Agent and after you press `ctrl+c` and there is no panic.

Agent shared component
- cmd/agent/subcommands/run/command.go
- pkg/cli/subcommands/check/command.go
- cmd/dogstatsd/subcommands/start/command.go

Platform integration:
- cmd/cluster-agent/subcommands/start/command.go
- cmd/cluster-agent-cloudfoundry/subcommands/run/command.go

Open telemetry:
- cmd/otel-agent/main.go

Security Agent:
- cmd/security-agent/subcommands/start/command.go

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]
LGTM
FYI looks like you missed some windows files

File: comp/aggregator/demultiplexer/demultiplexerimpl/demultiplexer.go [hunk 1]
```diff
@@ -74,6 +75,10 @@ func newDemultiplexer(deps dependencies) (provides, error) {
 	demultiplexer := demultiplexer{
 		AgentDemultiplexer: agentDemultiplexer,
 	}
+	deps.Lc.Append(fx.Hook{OnStop: func(ctx context.Context) error {
+		agentDemultiplexer.Stop(true)
+		return nil
+	}})
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply comp/aggregator/demultiplexer/demultiplexerimpl/demultiplexer.go:81]: Yes, indeed, that is why I added QA instructions and carefully check it is OK to stop the aggregator later.

**Review by vickenty — APPROVED** [review]

**Review by vboulineau — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Review by DylanLovesCoffee — APPROVED** [review]

File: pkg/epforwarder/epforwarder.go [hunk 2]
```diff
@@ -437,8 +437,10 @@ func (p *passthroughPipeline) Start() {
 }
 
 func (p *passthroughPipeline) Stop() {
-	p.strategy.Stop()
-	p.sender.Stop()
+	if p.strategy != nil {
+		p.strategy.Stop()
+		p.sender.Stop()
+	}
```
Comment (pgimalac) [inline pkg/epforwarder/epforwarder.go:443]: Is stopping `sender` only if `strategy` is not `nil` on purpose ?

**Review by pgimalac — APPROVED** [review]

File: pkg/epforwarder/epforwarder.go (hunk 2 above)
Comment (ogaca-dd) [reply pkg/epforwarder/epforwarder.go:443]: `sender` calls `Start` only if `p.strategy` is not nil: https://github.com/DataDog/datadog-agent/pull/22069/files/5499155b7a536ba122a2fa189d7eabf9236abbc5..f775c5fdbea2f103f3b6e6aa438ea3406659ede7#diff-6569a3f8ce509bc4602e1db6ae7933bbf5079532cbad740507d6dfe7a6fba3f4R433-R435

Comment (pgimalac) [reply pkg/epforwarder/epforwarder.go:443]: Ok perfect then, thanks for the reply !

**Pull request merged**